### PR TITLE
✨  [Frozen-deps]: Ignore testdata/ files

### PR DESCRIPTION
**Pull request by laurentsimon**
* **Please check if the PR fulfills these requirements**
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
bug fix

* **What is the current behavior?** (You can also link to an open issue here)
we don't ignore test files.

* **What is the new behavior (if this is a feature change)?**
we ignore test files.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no

* **Other information**:

### CODE REVIEW

**Review by naveensrinivasan — APPROVED** [review]

File: checks/frozen_deps.go [hunk 1]
```diff
@@ -55,12 +56,36 @@ func isDockerfilePinned(c *checker.CheckRequest) checker.CheckResult {
...
 }
 
+func isTestFile(pathfn string) bool {
+	d := path.Dir(pathfn)
+	if d == "testdata" {
+		return true
+	}
+
+	for d != "." {
+		r, err := path.Match("*/testdata", d)
```
Comment (azeemshaikh38) [inline checks/frozen_deps.go:66]: Should we be more generic here and say `path.Match("*test*")`?

File: checks/frozen_deps.go [hunk 2]
```diff
@@ -55,12 +56,36 @@ func isDockerfilePinned(c *checker.CheckRequest) checker.CheckResult {
 	return CheckFilesContent(frozenDepsStr, "*Dockerfile*", false, c, validateDockerfile)
 }
 
+func isTestFile(pathfn string) bool {
```
Comment (azeemshaikh38) [inline checks/frozen_deps.go:59]: Maybe simplify fn?

```
func isTestFile(pathfn string) bool {
	for d := path.Dir(pathfn); d != "."; d = path.Dir(d) {
		match, err := path.Match("*test*", d)
		if err != nil {
			return false
		}
		if match {
			return true
		}
	}
	return false
}
```

**Review by azeemshaikh38 — APPROVED** [review]
LGTM. Need to update all instances of `path` variable with something else. Otherwise import `pathlib "path"`.

File: checks/frozen_deps.go (hunk 1 above)
Comment (laurentsimon) [reply checks/frozen_deps.go:66]: I'm worried this is too broad and we'll have false negative. I was not convinced we should do this in the first place to be honest, but I wrote this as a quick fix to the failing `e2e` tests to unblock us. However: what if a repo has genuine docker files under a test folder, and the file is actually used for testing during CI/CD? We really don't want to miss these.
So I decided to remove the check here, and move it to `checkforcontent.go`:

1. Check the repo is scorecard repo
2. Check a folder is `testdata`

This way we only filter out our own repo's test files.

File: checks/frozen_deps.go (hunk 2 above)
Comment (laurentsimon) [reply checks/frozen_deps.go:59]: see new code.

File: checks/frozen_deps.go (hunk 1 above)
Comment (azeemshaikh38) [reply checks/frozen_deps.go]: Sounds good. This is a good example of what we were discussing yesterday - we need to find a way where clients can run Scorecards using a config. That allows clients to specify repo-specific files that Scorecard should ignore. Just something to keep in mind for future design discussions.

**Pull request merged**